ts/table-grid/table-grid.controller.ts

```diff
@@ -64,6 +64,7 @@ module Lui.Directives {
 				$scope.fixedRowDefinition = [];
 				$scope.scrollableHeaderRows = [];
 				$scope.scrollableRowDefinition = [];
+				$scope.allChecked = {value: false};
```

Comment:
required to use data-binding with checkboxes : http://stackoverflow.com/questions/18642371/checkbox-not-binding-to-scope-in-angularjs.
But, this property will be removed using your solution below.